### ISSUE TITLE
Use the real FlutterEngine

### ISSUE THREAD

**Issue by Clancey** [JeroMiya/xamarin.flutter#54]
IMHO in order for this to be successful, you need to continue to transpile the flutter SDK, but instead of re-writing the flutter engine in C# it should bind/pinvoke into the real engine.

**Comment by JeroMiya** [comment]
Yes, this is definately on the horizon to review and make a more informed decision on. Issues and questions that need to be resolved (not comprehensive):

- Can we emulate Dart's FFI functionality in C# efficiently or will we need to hand-rewrite all the bindings/dart-wrapping code to use P/Invoke directly?
- (Research TODO for me): what all functionality is in the engine to begin with? For example, is it functionality that is already implemented in Xamarin (e.g. platform integration, platform channels, etc...)? FFI wrappers for Skia?
- How to best translate Dart isolates to .Net/Xamarin. How do isolates interact with the Flutter engine?

I'd like to also review the UIWidgets implementation to see what their choices were (I suspect they rewrote the engine in C# and delegated to Unity):
https://github.com/UnityTech/UIWidgets
